### ISSUE TITLE
Handling attacks with post-selection decisions

### ISSUE THREAD

**Issue by irilyth** [buttonmen-dev/buttonmen#1143]
Some attacks require you to make some additional decisions after you select the dice that will participate in the attack, and before the attack is actually completed. (Fire and Turbo are two obvious examples.) Figure out how best to handle these, from a UI and/or backend perspective.

**Comment by irilyth** [comment]
Updating labels: I think this isn't a "question" any more; and, based on https://github.com/buttonmen-dev/buttonmen/issues/1143#issuecomment-50697848, it sounds like this is purely a UI thing.

**Comment by cgolubi1** [comment]
So, as Shadowshade suggests, it could be done as purely a UI thing, but it'd be a somewhat complicated UI thing in that case, since the UI would need to make one API call, check its result, and make another API call based on what it got from the first call.  It might not be too bad, and you might even be able to use some of the existing plumbing for form submission calls, but it could be a slight pain to debug.

**Comment by dwvanstone** [comment]
Here's an issue that (even with years and years of playing) I keep running into again with Fire dice. I suspect it's a newbie issue as well.

When I want to use the Fire die in a skill (or power) attack, I select the Fire die as part of the attack.

The response I get is "There is no valid attack corresponding to the dice selected."


I see Jota suggested a solution to this issue with a "checkbox for 'Use Fire dice to assist?'" option in issue https://github.com/buttonmen-dev/buttonmen/issues/1164. That would be a fine solution.

Another would be adding to the Attack menu:

  Power Attack
  Power Attack using Fire dice
  Skill Attack
  Skill Attack using Fire dice
  ...

**Comment by irilyth** [comment]
Yep, I think implementing the kind of interface that Jota and I were talking about here would (a) be less confusing for newbies (and experienced players); (b) obsolete a whole bunch of other issues with the Fire turndown page (by eliminating it); and (c) be a nice UI for Turbo Swing too, a skill that shows up on a bunch of (not yet available) buttons.

I don't know that anyone's opposed to it, I think it's just that no one's had the right combination of time and expertise to do it.

**Comment by blackshadowshade** [comment]
I'm tossing up how to implement the front-end proposed by @AdmiralJota, at least for Turbo Swing. At some stage, the UI needs to know whether a die has Turbo or not, and the question for me is whether we pass this information explicitly, or whether we parse the die recipe for an exclamation mark.

Also, Jota, do you happen to have any code samples / CSS lying around for your proposed front-end?

**Comment by blackshadowshade** [comment]
Okay, I've taken a look at the JSON passed when a Turbo die is present, and the snippet for activeDieArray looks like this:

    "activeDieArray":[
        {"value":1,"sides":4,"skills":[],"properties":[],"recipe":"(4)","description":"4-sided die"},
        {"value":11,"sides":12,"skills":[],"properties":[],"recipe":"(12)","description":"12-sided die"},
        {"value":20,"sides":20,"skills":[],"properties":[],"recipe":"(20)","description":"20-sided die"},
        {"value":3,"sides":7,"skills":["Turbo"],"properties":[],"recipe":"(X)!","description":"Turbo X Swing Die (with 7 sides)"}
    ]

So that looks pretty easy: any time a die with the Turbo skill is selected, pop up the Turbo section, otherwise hide it.
